ginivision/src/main/java/net/gini/android/vision/internal/fileimport/FileChooserActivity.java

```diff
@@ -144,6 +155,57 @@ protected void onActivityResult(final int requestCode, final int resultCode,
         finish();
     }
 
+    private boolean fileMatchesCriteria(final Intent data) {
```

Comment:
I would be happier if we would have an extensible and independent file checking solution. For "Open with" we would want to check the file in the Review Screen or the Analysis Screen. We might also add a new criteria or remove an existing one. For showing relevant error information to users we also would need what was wrong: too big, too many pages, ...

I drew up a little sketch of how that could look like:
![file-validation](https://user-images.githubusercontent.com/13149498/30696453-fca5c8b8-9edb-11e7-9a8b-3aa3ecdf252b.png)

`FileValidator` is used as the entry point. It checks each concrete validation in turn and could return the first error. Using the `Result` callers can check for success and use the `ValidationType` to handle each particular validation failure as required.

What do you think?

Reply:
I think that might be a little bit over engineered. Sure there can be a util class that can do the file checks, and returning true or false(and/or throwing an appropriate error).
The SDK is already pretty big for what it does and adding so much logic and files just to check some simple rules might be using a sledgehammer to crack a nut.

Reply:
It doesn't have to be like I drew it. Something smaller is also ok. It just should be stand alone so it can be used elsewhere, too. I wouldn't glue it to the file chooser. For the "Open with" we also need to do this file check.